### ISSUE TITLE
main slider : let flickity handle the lazyloading ?

### ISSUE THREAD

**Issue by ghost** [Nikeo/customizr#1239]
@eri-trabiccolo would there be any downside effect to apply to the main slider the same type of lazyload logic applied for the gallery carousel ( https://github.com/presscustomizr/customizr/commit/22c32ebf1eb7fdfbb34fb7a77ae40df593d0e2a4 ) 
 ?

**Comment by ghost** [comment]
The idea is of course to improve the performance by saving a lot of kbytes on page load, which is not the case right now because we smartload all the images of the slider.
=> flickity will only "smartload" <=> lazyload the first one

**Comment by eri-trabiccolo** [comment]
Wait I'm not sure I get what you mean:
1) _The idea is of course to improve the performance by saving a lot of kbytes on page load, which is not the case right now because we smartload all the images of the slider._
Main slider images are not smartloaded at all at the moment :) 
Did you mean the opposite?

2) In your commit, if you exclude the gallery imgs from the "theme smartload" why do we want to keep this:
`apply_filters( 'czr_thumb_html', //<- to allow the img smartload` ?
you know, why parsing them via PHP for a theme smartload which will not happen? (for prevdem?)

3) I don't see any big issue about this except for the responsive images:
Remember that we introduced the responsive images for the main slider just before releasing v4. In your commit above I see you unset the srcset (we should set to false sizes too then!?, sizes without srcset causes html validation error if I'm not wrong, see how we treat this case here:
https://github.com/presscustomizr/customizr/blob/dev/core/_dev/_utils/fn-3-thumbnails.php#L84
)
So basically ok, we have the flickity lazyload but we lose the responsive images because the flickity lazyload isn't able to handle it?!
We have an option for the responsive images, that will be completely overridden by the carousels this way.
Would be better, at this point, to implement our carousels lazyload js.

What do you think?

**Comment by ghost** [comment]
1. OK. I thought they were smartloaded.
2. That's right we don't need that in the normal case. But I wanted to keep it for the moment if for some reasons we need to deactivate the flickity lazyload ( by setting czr_lazy_load_img_in_gallery_carousels to false )
3. I'm doing some tests

**Comment by eri-trabiccolo** [comment]
I see, well historically they've never been smartloaded... why?
Not sure why, probably just because in the classical style I they couldn't:
not active slides are not displayed (display:none) so that jQuery isn't able to understand their position, hence the smartload wouldn't be triggered for them. And we never implemented any workaround for that.
But that's mostly a guess on the decision made several years ago :D 

That's not true for the flickity slider, as the non active slides are still displayed, that's why the theme smartload works fine for the modern gallery carousels.

**Comment by eri-trabiccolo** [comment]
Just to be clear about 3.
the way the slider responsive images are implemented for the main slider, as you know, is not that correct, because we load all the image sizes for a given image, which is not good, because only responsive versions of that image should be used.
I think we should open a different issue for that.

**Comment by ghost** [comment]
yes I know, this is always the same problem with srcset. And yes there should be an issue opened for that. Can you do it ?

**Comment by eri-trabiccolo** [comment]
Yes, I can, but I don't think it's a problem with the srcset, it's a problem with the way we want to use it :D 
It does what it has to do, but we want it doing something else :D
